test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/CredentialsTests.kt

```diff
@@ -45,68 +49,82 @@ class CredentialsTests {
+                    AuthenticationProvider.JWT -> Credentials.jwt("")
+                }
+            }
+        }
+
+        // Test Email/Password as a special case, due to it having two arguments
+        assertFailsWith<IllegalArgumentException> { Credentials.emailPassword("", "password") }
+        assertFailsWith<IllegalArgumentException> { Credentials.emailPassword("foo@bar.com", "") }
+
+        // Test Google as a special case as two types of Google login exists
+        assertFailsWith<IllegalArgumentException> { Credentials.google("", GoogleAuthType.AUTH_CODE) }
+        assertFailsWith<IllegalArgumentException> { Credentials.google("", GoogleAuthType.ID_TOKEN) }
     }
 
-    fun anonymous(): Credentials {
-        val creds = Credentials.anonymous()
+    private fun anonymous(): Credentials {
+        val creds: Credentials = Credentials.anonymous()
         // assertTrue(creds.asJson().contains("anon-user")) // Treat the JSON as an opaque value.
         return creds
     }
 
-//    @Test
-//    fun apiKey() {
-//        val creds = Credentials.apiKey("token")
-//        assertEquals(Credentials.Provider.API_KEY, creds.identityProvider)
-//        assertTrue(creds.asJson().contains("token")) // Treat the JSON as an opaque value.
-//    }
-//
-//    @Test
-//    fun apiKey_invalidInput() {
-//        assertFailsWith<IllegalArgumentException> { Credentials.apiKey("") }
-//        assertFailsWith<IllegalArgumentException> { Credentials.apiKey(TestHelper.getNull()) }
-//    }
-//
-//    @Test
-//    fun apple() {
-//        val creds = Credentials.apple("apple-token")
-//        assertEquals(Credentials.Provider.APPLE, creds.identityProvider)
-//        assertTrue(creds.asJson().contains("apple-token")) // Treat the JSON as a largely opaque value.
-//    }
-//
-//    @Test
-//    fun apple_invalidInput() {
-//        assertFailsWith<IllegalArgumentException> { Credentials.apple("") }
-//        assertFailsWith<IllegalArgumentException> { Credentials.apple(TestHelper.getNull()) }
-//    }
-//
-//    @Test
-//    fun customFunction() {
-//        val mail = TestHelper.getRandomEmail()
-//        val id = 666 + TestHelper.getRandomId()
-//        val creds = mapOf(
-//            "mail" to mail,
-//            "id" to id
-//        ).let { Credentials.customFunction(Document(it)) }
-//        assertEquals(Credentials.Provider.CUSTOM_FUNCTION, creds.identityProvider)
-//        assertTrue(creds.asJson().contains(mail))
-//        assertTrue(creds.asJson().contains(id.toString()))
-//    }
-//
-//    @Test
-//    fun customFunction_invalidInput() {
-//        assertFailsWith<IllegalArgumentException> { Credentials.customFunction(null) }
-//    }
+    private fun apiKey(): Credentials {
+        val creds: Credentials = Credentials.apiKey("token")
+        // assertTrue(creds.asJson().contains("token")) // Treat the JSON as an opaque value.
```

Comment:
why commented?

Reply:
It is waiting for https://github.com/realm/realm-core/issues/5348, I have added a comment in the code about it